docs/usage/getting-started/private-packages.md

```diff
@@ -259,6 +259,12 @@ However be aware that if your `.npmrc` is too big to encrypt then the above comm
 }
 ```
 
+#### Yarn 2+
+
+Renovate doesn't support reading `npmRegistries` and `npmScopes` from `.yarnrc.yml`, so `hostRules` (or `npmToken`) and `npmrc` should be configured like above.
+Renovate updates `npmRegistries` in `.yarnrc.yml` with resolved `hostRules` before running Yarn.
+For Renovate to overwrite existing `npmRegistries` entry, the key should match the `matchHost` without the protocol (`http:` or `https:`) and with the trailing slash.
```

Comment:
@rarkins Are we sure about this behavior?

Reply:
I didn't double check it but assumed @ylemkimon checked the source :)

Reply:
I've verified, this won't work for npm datasource. The hostrule won't applied, as it's not a url or a hostname.

Reply:
I think we're talking about different things here. https://github.com/renovatebot/renovate/blob/62a540c272d4f0b2738781d0442d61fd946cb482/lib/util/url.ts#L49-L55 https://github.com/renovatebot/renovate/blob/62a540c272d4f0b2738781d0442d61fd946cb482/lib/manager/npm/post-update/rules.ts#L20-L21

I meant following Renovate configuration:
```json
{
  "hostRules": [
    {
      "matchHost": "https://npm.pkg.github.com/",
      "hostType": "npm",
      "encrypted": {
        "token": "<Encrypted PAT token>"
      }
    }
  ]
}
```
will update `.yarnrc.yml` as following:
```yaml
npmRegistries:
  //npm.pkg.github.com/:
    npmAuthToken: <Decrypted PAT token>
```

Reply:
So `without` needs to be `with` `https?` prefix. I think the sample should be added too.

@ylemkimon can you create a new PR for it?

Reply:
I've opened #11696.